File: Zamba/ZambaWeb2013/zamba2010/Zamba.WFActivity.Regular/Action/DoChangeExpireDate.vb
Imports Zamba.Data
Imports Zamba.Core
Imports System.Xml.Serialization


''' -----------------------------------------------------------------------------
''' Project	 : Zamba.Business
''' Class	 : Core.DoChangeExpireDate
''' 
''' -----------------------------------------------------------------------------
''' <summary>
''' Regla que permite modificar la fecha de vencimiento de una tarea
''' </summary>
''' <remarks>
''' Hereda de WFRuleParent
''' </remarks>
''' <history>
''' 	[Hernan]	29/05/2006	Created
''' </history>
''' -----------------------------------------------------------------------------
<RuleCategory("Tareas"), RuleDescription("Modificar Vencimiento"), RuleHelp("Permite modificar el vencimiento de la tarea"), RuleFeatures(False)> <Serializable()> _
Public Class DoChangeExpireDate
    Inherits WFRuleParent
    Implements IDoChangeExpireDate
    Public Overrides Sub Dispose()

    End Sub
    Private _isLoaded As Boolean
    Private _isFull As Boolean
    Public Overrides ReadOnly Property IsFull() As Boolean
        Get
            Return _isFull
        End Get
    End Property
    Public Overrides ReadOnly Property IsLoaded() As Boolean
        Get
            Return _isLoaded
        End Get
    End Property
    Public Overrides Sub FullLoad()

    End Sub
    Public Overrides Sub Load()

    End Sub
    'TODO Martin, completar los parametros

    '''' -----------------------------------------------------------------------------
    '''' <summary>
    '''' 
    '''' </summary>
    '''' <param name="Id"></param>
    '''' <param name="Name"></param>
    '''' <param name="WFStep"></param>
    '''' <param name="Direccion1"></param>
    '''' <param name="Direccion2"></param>
    '''' <param name="Direccion3"></param>
    '''' <param name="Direccion4"></param>
    '''' <param name="Direccion5"></param>
    '''' <param name="Value1"></param>
    '''' <param name="Value2"></param>
    '''' <param name="Value3"></param>
    '''' <param name="Value4"></param>
    '''' <param name="Value5"></param>
    '''' <remarks>
    '''' </remarks>
    '''' <history>
    '''' 	[Hernan]	29/05/2006	Created
    '''' </history>
    '''' -----------------------------------------------------------------------------
    Public Sub New(ByVal Id As Int64, ByVal Name As String, ByVal wfstepId As Int64, ByVal Direccion1 As Int32, ByVal Direccion2 As Int32, ByVal Direccion3 As Int32, ByVal Direccion4 As Int32, ByVal Direccion5 As Int32, ByVal Value1 As Int32, ByVal Value2 As Int32, ByVal Value3 As Int32, ByVal Value4 As Int32, ByVal Value5 As Int32)
        MyBase.New(Id, Name, wfstepId) ', ListofRules.DoChangeExpireDate)
        Me._Direccion1 = Direccion1
        Me._Direccion2 = Direccion2
        Me._Direccion3 = Direccion3
        Me._Direccion4 = Direccion4
        Me._Direccion5 = Direccion5
        Me._Value1 = Value1
        Me._Value2 = Value2
        Me._Value3 = Value3
        Me._Value4 = Value4
        Me._Value5 = Value5
    End Sub
    '--ITEMS--
    'StateId=0

    'Properties
    Public Property Direccion1() As Int32 Implements IDoChangeExpireDate.Direccion1
        Get
            Return _Direccion1
        End Get
        Set(ByVal value As Int32)
            _Direccion1 = value
        End Set
    End Property
    Private _Direccion1 As Int32
    Public Property Direccion2() As Int32 Implements IDoChangeExpireDate.Direccion2
        Get
            Return _Direccion2
        End Get
        Set(ByVal value As Int32)
            _Direccion2 = value
        End Set
    End Property
    Private _Direccion2 As Int32
    Public Property Direccion3() As Int32 Implements IDoChangeExpireDate.Direccion3
        Get
            Return _Direccion3
        End Get
        Set(ByVal value As Int32)
            _Direccion3 = value
        End Set
    End Property
    Private _Direccion3 As Int32
    Public Property Direccion4() As Int32 Implements IDoChangeExpireDate.Direccion4
        Get
            Return _Direccion4
        End Get
        Set(ByVal value As Int32)
            _Direccion4 = value
        End Set
    End Property
    Private _Direccion4 As Int32
    Public Property Direccion5() As Int32 Implements IDoChangeExpireDate.Direccion5
        Get
            Return _Direccion5
        End Get
        Set(ByVal value As Int32)
            _Direccion5 = value
        End Set
    End Property
    Private _Direccion5 As Int32

    Public Property Value1() As Int32 Implements IDoChangeExpireDate.Value1
        Get
            Return _Value1
        End Get
        Set(ByVal value As Int32)
            _Value1 = value
        End Set
    End Property
    Private _Value1 As Int32
    Public Property Value2() As Int32 Implements IDoChangeExpireDate.Value2
        Get
            Return _Value2
        End Get
        Set(ByVal value As Int32)
            _Value2 = value
        End Set
    End Property
    Private _Value2 As Int32
    Public Property Value3() As Int32 Implements IDoChangeExpireDate.Value3
        Get
            Return _Value3
        End Get
        Set(ByVal value As Int32)
            _Value3 = value
        End Set
    End Property
    Private _Value3 As Int32
    Public Property Value4() As Int32 Implements IDoChangeExpireDate.Value4
        Get
            Return _Value4
        End Get
        Set(ByVal value As Int32)
            _Value4 = value
        End Set
    End Property
    Private _Value4 As Int32
    Public Property Value5() As Int32 Implements IDoChangeExpireDate.Value5
        Get
            Return _Value5
        End Get
        Set(ByVal value As Int32)
            _Value5 = value
        End Set
    End Property
    Private _Value5 As Int32

    '[Ezequiel] 09/06/2009 - Se elimino el Try-Catch
    Public Overloads Overrides Function Play(ByVal results As System.Collections.Generic.List(Of ITaskResult)) As System.Collections.Generic.List(Of ITaskResult)
        Dim playRule As New Zamba.WFExecution.PlayDoChangeExpireDate()
        Return playRule.Play(results, Me)
    End Function
    Public Overloads Overrides Function Playweb(ByVal results As System.Collections.Generic.List(Of ITaskResult), ByRef RulePendingEvent As RulePendingEvents, ByRef ExecutionResult As RuleExecutionResult, ByRef Params As Hashtable) As System.Collections.Generic.List(Of ITaskResult)
        Dim playrule As New WFExecution.PlayDoChangeExpireDate()
        Return playrule.Play(results, Me)
    End Function
    'Protected Overloads Overrides Function Play(ByVal Result As TaskResult) As TaskResult
    '    Try
    '        If IsNothing(result.ExpireDate) = False Then
    '            Dim NuevaFecha As DateTime = result.ExpireDate

    '            Dim Minutos As Int32 = Value1
    '            Dim Horas As Int32 = Value2
    '            Dim Dias As Int32 = Value3
    '            Dim Semanas As Int32 = Value4
    '            Dim Meses As Int32 = Value5

    '            If Direccion1 = 1 Then
    '                Minutos = -Minutos
    '            End If
    '            If Direccion2 = 1 Then
    '                Horas = -Horas
    '            End If
    '            If Direccion3 = 1 Then
    '                Dias = -Dias
    '            End If
    '            If Direccion4 = 1 Then
    '                Semanas = -Semanas
    '            End If
    '            If Direccion5 = 1 Then
    '                Meses = -Meses
    '            End If

    '            NuevaFecha = NuevaFecha.AddMinutes(Minutos)
    '            NuevaFecha = NuevaFecha.AddHours(Horas)
    '            NuevaFecha = NuevaFecha.AddDays(Dias)
    '            NuevaFecha = NuevaFecha.AddDays(Semanas * 7)
    '            NuevaFecha = NuevaFecha.AddMonths(Meses)

    '            WFBusiness.ChangeExpireDateTask(result, NuevaFecha)
    '        End If
    '    Catch ex As Exception
    '       zamba.core.zclass.raiseerror(ex)
    '    End Try
    'End Function

End Class

File: Zamba/ZambaWeb2013/zamba2010/Zamba.WorkFlow.SQLExecution/Zamba.WorkFlow.Execution.ActivityLibrary/Activity/DoChangeExpireDate.cs
using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Reflection;
using System.Diagnostics;
using Zamba.Core;
//using Zamba.WFUserControl;

namespace Zamba.WFActivity.Xoml
{
    
    /// <summary>
    /// Regla que permite modificar la fecha de vencimiento de una tarea
    /// </summary>
    [RuleDescription("VENCIMIENTO DE TAREA"), RuleHelp("Modifica la fecha de vencimiento de una tarea.")]
    public partial class DoChangeExpireDate : ZRule, IResultActivity, IDoChangeExpireDate
	{

        private Int32 _Direccion1, _Direccion2, _Direccion3, _Direccion4, _Direccion5;
        private Int32 _Value1, _Value2, _Value3, _Value4, _Value5;


		public DoChangeExpireDate()
		{
			InitializeComponent();
		}

        protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
        {
            Zamba.WFExecution.PlayDoChangeExpireDate play =
                new Zamba.WFExecution.PlayDoChangeExpireDate();
            play.Play(Results, this);
            return ActivityExecutionStatus.Closed;
        }

        public override void SetParams(object[] Params)
        {
            Direccion1 = Int32.Parse(Params.GetValue(0).ToString());
            Direccion2 = Int32.Parse(Params.GetValue(1).ToString());
            Direccion3 = Int32.Parse(Params.GetValue(2).ToString());
            Direccion4 = Int32.Parse(Params.GetValue(3).ToString());
            Direccion5 = Int32.Parse(Params.GetValue(4).ToString());
            Value1 = Int32.Parse(Params.GetValue(5).ToString());
            Value2 = Int32.Parse(Params.GetValue(6).ToString());
            Value3 = Int32.Parse(Params.GetValue(7).ToString());
            Value4 = Int32.Parse(Params.GetValue(8).ToString());
            Value5 = Int32.Parse(Params.GetValue(9).ToString());
        }
        public override System.Collections.Generic.List<object> GetParams()
        {
            System.Collections.Generic.List<object> Params = new System.Collections.Generic.List<object>();
            Params.Add(Direccion1);
            Params.Add(Direccion2);
            Params.Add(Direccion3);
            Params.Add(Direccion4);
            Params.Add(Direccion5);
            Params.Add(Value1);
            Params.Add(Value2);
            Params.Add(Value3);
            Params.Add(Value4);
            Params.Add(Value5);
            return Params;
        }

        #region IDoChangeExpireDate Members

        public int Direccion1
        {
            get
            {
                return _Direccion1;
            }
            set
            {
                _Direccion1 = value;
            }
        }

        public int Direccion2
        {
            get
            {
                return _Direccion2;
            }
            set
            {
                _Direccion2 = value;
            }
        }

        public int Direccion3
        {
            get
            {
                return _Direccion3;
            }
            set
            {
                _Direccion3 = value;
            }
        }

        public int Direccion4
        {
            get
            {
                return _Direccion4;
            }
            set
            {
                _Direccion4 = value;
            }
        }

        public int Direccion5
        {
            get
            {
                return _Direccion5;
            }
            set
            {
                _Direccion5 = value;
            }
        }

        public int Value1
        {
            get
            {
                return _Value1;
            }
            set
            {
                _Value1 = value;
            }
        }

        public int Value2
        {
            get
            {
                return _Value2;     
            }
            set
            {
                _Value2 = value;
            }
        }

        public int Value3
        {
            get
            {
                return _Value3;
            }
            set
            {
                _Value3 = value;
            }
        }

        public int Value4
        {
            get
            {
                return _Value4;
            }
            set
            {
                _Value4 = value;
            }
        }

        public int Value5
        {
            get
            {
                return _Value5;
            }
            set
            {
                _Value5 = value;
            }
        }

        #endregion
     }
}
